app/assets/javascripts/controllers/middleware_topology/middleware_topology_controller.js

```diff
@@ -112,8 +150,25 @@ function MiddlewareTopologyCtrl($scope, $http, $interval, $location, topologySer
             default :
                 return { x: -9, y: -9, height: 18, width: 18, r: 17};
         }
+    }
 
-    };
+    self.getTextPosition = function getTextPosition(d) {
```

Comment:
It seems like this is an enhancement to place better the labels on the graph - which is great, but not related to adding datasources to topology. I suggest submitting a separate PR with this. worth to add the defaults in https://github.com/ManageIQ/manageiq/blob/master/app/assets/javascripts/services/topology_service.js
And then it can influence all the topologies if needed, not only middleware topology.